### PR TITLE
[script] [common-arcana] Remove duplicate code to set moon data; use new util methods

### PR DESCRIPTION
**Pull request by KatoakDR**
### Background
* https://github.com/rpherbig/dr-scripts/pull/5408 introduced new methods to encapsulate logic parsing `UserVars.moons` to know if any are visible and which ones are

### Changes
* 🧹 Clean up where code can be replaced by calls to the new util methods

### CODE REVIEW

File: common-arcana.lic [hunk 1]
```diff
@@ -560,21 +560,20 @@ module DRCA
...
+    echo "No planet available to cast #{data['abbrev']}"
     nil
   end
 
   def set_moon_data(data)
     return data unless data['moon']
 
-    check_moonwatch
-    # Look for a moon that will be up for the next few minutes
-    moon = UserVars.moons.find { |moon_name, moon_data| UserVars.moons['visible'].include?(moon_name) && moon_data['timer'] >= 4 }.first
```
Comment (KatoakDR) [inline common-arcana.lic:572]: The new `visible_moons` method handles parsing  `UserVars.moons` and calling `check_moonwatch`

File: common-arcana.lic [hunk 2]
```diff
@@ -804,26 +803,13 @@ module DRCA
...
-        if weather =~ /inside/
-          echo "*** You're inside and there are no available moons. You're going to have a hard time casting #{data['abbrev']}"
-        end
-
-        unless moon = UserVars.moons['visible'].first
-          return
-        end
-        data['cast'] = "cast #{moon}"
-      end
-    end
```
Comment (KatoakDR) [inline common-arcana.lic:825]: `set_moon_data` does these checks and updates the cast command

There is no need for the weather check, it's immaterial to any decisions being made

File: common-arcana.lic [hunk 3]
```diff
@@ -560,21 +560,20 @@ module DRCA
       data['cast'] = "cast #{cast_on}"
       return data
     end
-    echo 'Could not find any planets to cast on'
+    echo "No planet available to cast #{data['abbrev']}"
     nil
   end
 
   def set_moon_data(data)
```
Comment (rpherbig) [inline common-arcana.lic:567]: This probably isn't something we can change here, but the Ruby convention would be to name the function `set_moon_data!`

Comment (KatoakDR) [reply common-arcana.lic:567]: Good to know, thanks

**Pull request merged**